shuffle/cli/src/account.rs

```diff
@@ -26,20 +28,48 @@ use std::{
     io,
     path::{Path, PathBuf},
 };
+use url::Url;
 
 // Creates new account from randomly generated private/public key pair.
-pub fn handle(home: &Home, root: Option<PathBuf>, network: Network) -> Result<()> {
+pub async fn handle(home: &Home, root: Option<PathBuf>, network: Network) -> Result<()> {
     let network_home = home.new_network_home(&network.get_name());
     network_home.generate_paths_if_nonexistent()?;
     check_nodeconfig_exists_if_localhost_used(home, &network)?;
 
-    if network_home.get_latest_account_path().exists() {
+    if network_home.get_latest_account_key_path().exists() {
         match user_wants_another_key(&network_home) {
             true => archive_current_files_in_latest(&network_home)?,
             false => return Ok(()),
         }
     }
+    let new_account = generate_new_account(&network_home)?;
+    let test_account = generate_test_account(&network_home)?;
+
+    //creating account onchain based on network provided by user
+    match network.get_name().as_str() {
+        LOCALHOST_NAME => handle_account_creation_on_localhost(
+            home,
+            &network_home,
+            &network,
+            root,
+            &new_account,
+            &test_account,
+        ),
+        TROVE_TESTNET_NETWORK_NAME => {
+            handle_account_creation_on_network(&network, &new_account, &test_account).await
+        }
+        _ => Ok(()),
+    }
+}
 
+fn handle_account_creation_on_localhost(
+    home: &Home,
+    network_home: &NetworkHome,
+    network: &Network,
+    root: Option<PathBuf>,
+    new_account: &LocalAccount,
+    test_account: &LocalAccount,
+) -> Result<()> {
```

Comment:
This method doesn't add much tbh, and the naming is confusing because it's actually creating two accounts. Why not just invoke the two lines directly in the match above?

Reply:
removing and putting it's contents inside of match statement for LOCALHOST_NAME